packages/wallet/src/lib/utils/proof/errors.ts

```diff
@@ -0,0 +1,38 @@
+export class NotAProofLinkError extends Error {
```

Comment:
on peut pas faire un factory?

Reply:
?

Reply:
La seule chose qui change dans le code c'est le "name". Donc faire une factory qui prend en param le name

export class ProofKeyNotFoundError extends Error {
  constructor(name: string, message: string) {
    super(message);
    this.name = name;
    Object.setPrototypeOf(this, ProofKeyNotFoundError.prototype);
  }
}

// Factory function
export function createProofKeyNotFoundError(name: string, message: string): ProofKeyNotFoundError {
  return new ProofKeyNotFoundError(name, message);
}

Reply:
il n'y a pas moyen de factoriser ce code?

Reply:
ce ne sont que des constructeurs, pas de sens pou moi. L'idée c'est de pouvoir utiliser instanceof pour discriminer les erreurs dans un try catch